packages/kbn-ftr-common-functional-ui-services/services/toasts.ts

```diff
@@ -0,0 +1,154 @@
+  private readonly defaultFindTimeout = this.config.get('timeouts.find');
+  /**
+   * Returns the title and message of a specific error toast.
+   * This method is specific to toasts created via `.addError` since they contain
+   * an additional button, that should not be part of the message.
+   *
+   * @param index The index of the toast (1-based, NOT 0-based!) of the toast. Use first by default.
+   * @param titleOnly If this is true, only the title of the error message is returned. There are error messages that only contain a title, no message.
+   * @returns The title and message of the specified error toast.https://github.com/elastic/kibana/issues/17087
+   */
+  public async getErrorToastByIndex(
+    index: number = 1,
+    titleOnly: boolean = false
+  ): Promise<{ title: string; message?: string }> {
+    const title = await this.getToastTitleByIndex(index);
+    if (titleOnly) return { title };
+
+    const toast = await this.getToastElementByIndex(index);
+    const messageElement = await this.testSubjects.findDescendant('errorToastMessage', toast);
+    const message: string = await messageElement.getVisibleText();
+
+    return { title, message };
+  }
+
+  public async toastMessageByTestSubj(testSubj = 'csp:toast-success') {
+    const testSubjSvc = this.testSubjects;
+    return {
+      async getElement(): Promise<WebElementWrapper> {
+        return await testSubjSvc.find(testSubj);
+      },
+      async clickToastMessageLink(linkTestSubj = 'csp:toast-success-link') {
+        const element = await this.getElement();
+        const link = await element.findByTestSubject(linkTestSubj);
+        await link.click();
+      },
+    };
+  }
+
+  /**
+   * Dismiss a specific toast from the toast list. Since toasts usually should time out themselves,
+   * you only need to call this for permanent toasts (e.g. error toasts).
+   *
+   * @param index The 1-based index of the toast to dismiss. Use first by default.
+   */
+  public async dismissToastByIndex(index: number = 1): Promise<void> {
+    const toast = await this.getToastElementByIndex(index);
+    await toast.moveMouseTo();
+    const dismissButton = await this.testSubjects.findDescendant('toastCloseButton', toast);
+    await dismissButton.click();
+  }
+
+  public async closeToastIfExists(): Promise<void> {
+    const toastShown = await this.find.existsByCssSelector('.euiToast');
+    if (toastShown) {
+      try {
+        await this.testSubjects.click('toastCloseButton');
+      } catch (err) {
+        // ignore errors, toast clear themselves after timeout
+      }
+    }
+  }
+
+  public async closeToast(): Promise<string> {
+    const toast = await this.find.byCssSelector('.euiToast', 6 * this.defaultFindTimeout);
+    await toast.moveMouseTo();
+    const title = await (await this.testSubjects.find('euiToastHeader__title')).getVisibleText();
+
+    await this.testSubjects.click('toastCloseButton');
+    return title;
+  }
```

Comment:
I can see many calls like `await pageObjects.common.closeToast();` and this call is slowed done by extra WEbDriver calls to return title text:

I suggest closeToast to only close it:

```
public async closeToast(): Promise<void> {
    const toast = await this.find.byCssSelector('.euiToast', 6 * this.defaultFindTimeout);
    await toast.moveMouseTo();
    await this.testSubjects.click('toastCloseButton');
  }
```

For the case when we want to close and still get text, let's rename the current method to `getTitleAndClose()`;